### PR TITLE
Nuke: Enhance Viewer process override with colorspace, display-view

### PR DESCRIPTION
**Pull request by jakubjezek001**
## Changelog Description
This pull request addresses the limitations of the Baking preset **Viewer process override** input field. Previously, the field was restricted to accepting only display and view type names for presets, leading to potential confusion among users. With this update, we have expanded the functionality to include colorspace names, thereby enhancing the user experience and aligning with industry-standard practices.

 
### Key Features Implemented:
*   Extended input validation to accept colorspace names in addition to display and view type names.
*   Integrated anatomy template formatting support, allowing names to be tied to specific contexts within the pipeline.
*   Ensured compatibility with upcoming OCIO config publications that will feature baked-in looks such as LMT (Look Modification Transform), ODT (Output Device Transform), and Display & View transforms, where naming conventions are context-specific.
*   Revised settings architecture for Viewer/Monitor and Baking color space targets.
*   Implemented settings conversion from legacy structure for the copy settings and project overrides tool.

The implementation has been thoroughly tested to confirm that it meets the desired specifications and behaves as expected across various scenarios. This enhancement is a step forward in our ongoing efforts to streamline our color management workflow and provide robust tools for our users.

## Additional info
Some debug logging were removed and f-string conversions done

## Testing notes:
To replicate this test case, ensure that the configuration with context-specific names is loaded within a Nuke task session. Reach out on Discord to coordinate a joint testing effort.

### CODE REVIEW

**Review by moonyuet — APPROVED** [review]
I tested with/without the colorspace override. Extract review can publish accordingly with or without colorspace override. When I loaded the mov back to the scene, the video appears to show my preferences of display and view with the colorspace override. It also stays with default Rec709 which can be set in baking target colorspace.
 
image

The left green read node is the one which publishes without enabling colorspace override, and the right one is with colorspace override enabled and setting the display as ACES, views as sRGB.

File: server_addon/nuke/client/ayon_nuke/api/lib.py [hunk 1]
```diff
@@ -1623,21 +1608,25 @@ def set_root_colorspace(self, imageio_host):
...
             workfile_settings["monitor_lut"]
         )
-        monitor_lut_data.update({
-            "workingSpaceLUT": workfile_settings["working_space"],
-            "int8Lut": workfile_settings["int_8_lut"],
-            "int16Lut": workfile_settings["int_16_lut"],
-            "logLut": workfile_settings["log_lut"],
-            "floatLut": workfile_settings["float_lut"]
-        })
+        workfile_settings_output |= monitor_lut_data
```
Comment (iLLiCiTiT) [inline server_addon/nuke/client/ayon_nuke/api/lib.py:1617]:
```suggestion
        workfile_settings_output.update(monitor_lut_data)
```

File: server_addon/nuke/client/ayon_nuke/api/plugin.py [hunk 2]
```diff
@@ -638,12 +638,18 @@ def get_imageio_baking_profile(self):
...
 
-        # TODO: this is only securing backward compatibility lets remove
-        # this once all projects's anatomy are updated to newer config
-        if "baking" in nuke_imageio.keys():
-            return nuke_imageio["baking"]["viewerProcess"]
+        if (
+            "baking_target" in nuke_imageio.keys()
+            and nuke_imageio["baking_target"]["enabled"]
+        ):
+            return nuke_imageio["baking_target"]
```
Comment (iLLiCiTiT) [inline server_addon/nuke/client/ayon_nuke/api/plugin.py:645]:
```suggestion
        if nuke_imageio["baking_target"]["enabled"]:
            return nuke_imageio["baking_target"]
```

**Review by iLLiCiTiT — APPROVED** [review]
LGTM

**Pull request merged**